### DISCUSSION TITLE
Composite key made up of two composite foreign keys?

### DISCUSSION THREAD

**Discussion by samanthafong** [supabase/supabase#27355] [Questions]
Hi, I'm a database noob trying to build a work management app with a table logging each user's long-term goals, and another table logging each user's medium-term projects. User's Goals table has a composite primary key consisting of user ID and project name, and I don't want to use a surrogate key because I don't want a user to be able to have multiple goals with the same name. Same setup for User's Projects.

Goals and Projects will have a many-to-many relationship as a project can belong to more than one goal, and vice versa. Problem arises when trying to create lookup table to show the relationship between the goals and the projects. I want this lookup table to have a composite primary key so there cannot be more than one relationship between a certain goal and a certain project.  **So for this lookup table, I'm trying to create a composite primary key consisting of two composite primary keys.**  This is what I've attempted: 

![image](https://github.com/supabase/supabase/assets/47237413/d72e7422-5e51-419d-bf45-b06c2b79473b)

And I get the following error message:
![image](https://github.com/supabase/supabase/assets/47237413/97affdf0-fc37-4e1d-8f47-9bbe8d67b25c)

**Am I doing something wrong, or is a feature that hasn't been created yet?** 
Also, if there's an obvious alternate solution that I'm missing, would love if someone could point that out.

Many sincere thanks in advance for any insight!

**Comment by GaryAustin1** [comment]
Composite foreign keys need matching columns on both ends.  So you need two different columns not the same testgoal_id for both.
https://pysql.tecladocode.com/section08/lectures/01_composite_primary_keys/
